app/src/main/java/com/eis/phoneringer/structure/RingCommandHandler.java

```diff
@@ -49,12 +49,13 @@ public RingCommand parseMessage(SMSMessage smsMessage) {
 
     /**
      * Extracts the password and the peer from the RingCommand and creates a SMSMessage object
+     * The password is sent with the {@link #SPLIT_CHARACTER} in front
      *
      * @param ringCommand to parse, it must be a valid one
      * @return a SMSMessage object
      */
     public SMSMessage parseCommand(RingCommand ringCommand) {
-        return new SMSMessage(ringCommand.getPeer(), ringCommand.getPassword());
+        return new SMSMessage(ringCommand.getPeer(), SPLIT_CHARACTER + ringCommand.getPassword());
```

Comment:
I still don't get why this is called SPLIT_CHARACTER if it doesn't split anything, you could name it something like ring command prefix or whatever: what if we decide to have more commands and use this class to parse them? Think big

Reply:
It splits different fields to allow parsing of the message, right?

Reply:
But there are no _different fields_, there is just the password right now as field, the sms outcome is: "_password"

Reply:
lol, you're right

Reply:
Because in the parseMessage method the SMSMessage content is splitted thanks to the "_" character.
And thinks if we have ti send a command with the password AND Idk, the volume we want to play the ringtone: we are going to use 2 SPLIT_CHARACTER like that: _password_100 where 100 is the max volume

Reply:
I would modify parseMessage and make it use `[split()](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#split-java.lang.String-)` to separate content by looking for `SPLIT_CHARACTER` inbetween different elements

Reply:
Why don't check for _ as the first character and then get the password as substring(1)? You are not splitting anything as @CremaLuca already noted.

Reply:
As you wish.
I will change the name from "SPLIT_CHARACTER" to "COMMAND_IDENTIFIER"

Reply:
@giovannivelludo the method already uses `split(SPLIT_CHARACTER)`

Reply:
sorry, I'm retarded. then you should change
```
return new RingCommand(smsMessage.getPeer(), parts[1]);
```
to
```
return new RingCommand(smsMessage.getPeer(), parts[0]);
```
if you're removing the `_` character from before the password

Reply:
And if we are using our library for another application?

Reply:
Messages can get confused

Reply:
> And if we are using our library for another application?

What library?

Reply:
smslibrary

Reply:
I thought the purpose of `SPLIT_CHARACTER` was separating different fields in a message, not being an identifier of this application.  
If you want to use a custom code for this app you should use your own [`MessageParseStrategy`](https://github.com/Cogno-IDU/smslibrary/blob/master/smslibrary/src/main/java/com/eis/communication/MessageParseStrategy.java).  
My personal opinion is that this unnecessarily overcomplicates things, because it would be easier to do the parsing directly in the app like you did, without having to create an entire class that extends `MessageParseStrategy`, then calling `SMSMessageHandler.getInstance().setMessageParseStrategy()`, and then having to call `SMSMessageHandler`'s methods everytime you want to check if a message is meant for this app.
So I agree with using `SPLIT_CHARACTER` to identify messages meant for this app, but then I would change the name of that variable if separating fields is not its purpose.

Reply:
Take a look to this commit 918ead5

Reply:
Sorry, this conversation wasn't marked as resolved and the code on top of it was the old one